src/Battleships/Battleships.java

```diff
@@ -30,6 +30,8 @@ public static class GameState {
 
         private int currentPlayer;
 
+        private int winner = 0;
+
```

Comment:
Consider keeping Player objects as currentPlayer and winner instead of indices. If there is no player (this can 100% be the case for winner, because for most of the game there'll be no winner) then you could store null or use Java's Optional class.